services/server/server.go

```diff
@@ -517,33 +517,36 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 func (s *Server) validateSSHKey(ctx context.Context, sshKey string) error {
 	tempFile, err := ioutil.TempFile("", "temp_keys_*")
 	if err != nil {
-		return err
+		return errors.WithStack(err)
 	}
+	tempFile.Close()                 //nolint:errcheck
+	defer os.Remove(tempFile.Name()) //nolint:errcheck
```

Comment:
ioutil.TempFile returns open file. We should close it; otherwise, we will leak file descriptor.
We also did not removed that temporary file.